src/blockrelay/graphene.cpp

```diff
@@ -1197,7 +1202,9 @@ bool CGrapheneBlockData::CheckGrapheneBlockTimer(const uint256 &hash)
         // To make the timeout random we adjust the start time of the timer forward
         // or backward by a random amount plus or minus 0.2 seconds.
         FastRandomContext insecure_rand(false);
-        uint64_t nOffset = nTimeToWait - (800 + (insecure_rand.rand64() % 400) + 1);
+        uint64_t nStartInterval = nTimeToWait * 0.8;
+        uint64_t nIntervalLen = 2 * (nTimeToWait * 0.2);
+        uint64_t nOffset = nTimeToWait - (nStartInterval + (insecure_rand.rand64() % nIntervalLen) + 1);
```

Comment:
nOffset = nTimeToWait - (nStartInterval + (insecure_rand.rand64() % nIntervalLen) + 1)

nOffset = nTimeToWait - nStartInterval - (insecure_rand.rand64() % nIntervalLen) - 1

nOffset = nTimeToWait - nTimeToWait * 0.8 - (insecure_rand.rand64() % (nIntervalLen) - 1

nOffset = nTimeToWait * 0.2 - (insecure_rand.rand64() % nIntervalLen) - 1

nOffset = nTimeToWait * 0.2 - (insecure_rand.rand64() % (nTimeToWait * 0.4) ) - 1

split rand into extremes:
Largest:
nOffset = nTimeToWait * 0.2 - ((nTimeToWait * 0.4)-1 ) - 1
nOffset = nTimeToWait * 0.2 - (nTimeToWait * 0.4) - 2

nOffset = -0.2*nTimeToWait - 2

        ***^*** is negative, so std::make_pair(GetTimeMillis() + nOffset, false); will trigger right away.  

I think maybe you have mis-offset the logic.

Smallest:
nOffset = nTimeToWait * 0.2 - (0) - 1